### ISSUE TITLE
Discussion on Devfile volume size support

### ISSUE THREAD

**Issue by maysunfaisal** [openshift/odo#2511]
[kind/Feature]

## Which functionality do you think we should add?
Currently, a devfile https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/java-web-spring/devfile.yaml has no option to configure size for a volume.

The Devfile schema does not mention it either - https://github.com/redhat-developer/devfile/blob/master/docs/devfile.json

We should have an option to mention the volume size. On the odo contributor's meeting, we discussed possible avenues:
- size could be mentioned in the devfile under the `volumes` section
- size could be mentioned in a global preference file that takes affect for all the devfiles(this is a short term solution if the devfile proposed change is not accepted)

## Why is this needed?
When working for issue https://github.com/openshift/odo/issues/2472, noticed that if we want odo to adopt the devfile, there is currently no way to configure a volume's size. This is currently present as an option in config.yaml and will be nice to have when consuming the devfile instead of having a default `1Gi` like odo did previously.

**Comment by maysunfaisal** [comment]
@girishramnani Hi Girish, since we discussed in the odo contributor's call about the possibility of having the volume size in a preference file, out of curiosity does odo currently have a preference file?

**Comment by kadel** [comment]
> * size could be mentioned in the devfile under the `volumes` section
 
We need to make sure that this doesn't get lost, and size is added to Devfile v2 specification

> * size could be mentioned in a global preference file that takes affect for all the devfiles(this is a short term solution if the devfile proposed change is not accepted)

+1 We need this as for term solution before we can switch to devfile v2
